src/ftw/utils.py

```diff
@@ -36,3 +44,58 @@ def validate_checksums(checksum_file, root_directory):
             print("Checksum mismatch: {file_path}")
             return False
     return True
+
+
+def harvest_to_datetime(harvest_day: int, year: int) -> pd.Timestamp:
+    """
+    Convert a harvest integer (day of the year) to a datetime object.
+
+    Args:
+        harvest_day (int): Day of the year (1-365).
+        year (int): The year for which the date is to be calculated.
+
+    Returns:
+        pd.Timestamp: Corresponding datetime object.
+    """
+    return pd.to_datetime(f"{year}-{harvest_day}", format="%Y-%j")
+
+
+# to-do func to get harvest integer from user provided bbox
+def get_harvest_integer_from_bbox(
+    bbox: list[int],
+    start_year_raster_path: str = SUMMER_START_RASTER_PATH,
+    end_year_raster_path: str = SUMMER_END_RASTER_PATH,
+) -> list[int]:
+    """
+    Gets harvest integer from a user-provided bounding box. Note currently just uses summer crops.
+
+    Args:
+        bbox (str): Bounding box in the format 'minx,miny,maxx,maxy'.
+
+    Returns:
+        list: start and end harvest integer (day of the year).
+    """
+
+    start_harvest_dset = xr.open_dataset(start_year_raster_path, engine="rasterio")
+    end_harvest_dset = xr.open_dataset(end_year_raster_path, engine="rasterio")
+
+    # Clip the datasets to the bounding box
+    start_value = start_harvest_dset.rio.clip_box(
+        bbox[0], bbox[1], bbox[2], bbox[3], allow_one_dimensional_raster=True
+    )
+
+    if len(start_value["band_data"][0][0]) > 1:
+        start_value = int(np.mean(start_value["band_data"][0][0]))
```

Comment:
is this reasonable? or if the area submitted is very large do we want to preserve the exact days from the crop calendar and return a list of lists with the exact start/end day pairs?

Reply:
mean seems reasonable to me!  Agree it would be nice to log out when this happens.  Maybe log out mean + stdev

Reply:
I'm not overly deep into this, but is there a chance that at the edges mean would compute weird results? Especially thinking about the southern hemisphere.

Let's say you get 360 and 4, the mean of that would not be overly suitable.

Reply:
great call @m-mohr! I'll can do some analysis of the summer crop calendars to see how likely this scenario is to occur. I can also add a quick check that makes sure if the difference between multiple values in >x number of days (30? 45? not sure what would be a reasonable default), we don't take the mean, we could return multiple crop calendar values and the geometries that correspond to those values.

Reply:
@martham93 You could use the circular mean, i.e. translate the values into angles and then compute the mean, see https://en.wikipedia.org/wiki/Circular_mean or https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.circmean.html

Reply:
nice thanks for that suggestion! Circular Mean + Logging added in [d2139df](https://github.com/fieldsoftheworld/ftw-baselines/pull/121/commits/d2139dfdf2f9917d22c99262faf43cf73d5ce70a)